components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -161,31 +176,47 @@ export default class NotificationsTab extends React.PureComponent {
             stringKeys += ',' + this.state.customKeys;
         }
 
-        data.mention_keys = stringKeys;
-        data.first_name = this.state.firstNameKey.toString();
-        data.channel = this.state.channelKey.toString();
+        const data: any = {
+            email: this.state.enableEmail,
+            desktop_sound: this.state.desktopSound,
+            desktop: this.state.desktopActivity,
+            push: this.state.pushActivity,
+            push_status: this.state.pushStatus,
+            comments: this.state.notifyCommentsLevel,
+            mark_unread: this.state.markUnread,
+            first_name: this.state.firstNameKey ? 'true' : 'false',
+            channel: this.state.channelKey ? 'true' : 'false',
+            mention_keys: stringKeys,
+        };
+
+        if (!data.auto_responder_message || data.auto_responder_message === '') {
+            data.auto_responder_message = Utils.localizeMessage(
+                'user.settings.notifications.autoResponderDefault',
+                'Hello, I am out of office and unable to respond to messages.'
+            );
+        }
 
         this.setState({isSaving: true});
 
-        this.props.actions.updateMe({notify_props: data}).
-            then(({data: result, error: err}) => {
-                if (result) {
-                    this.handleUpdateSection('');
-                    this.setState(getNotificationsStateFromProps(this.props));
-                } else if (err) {
-                    this.setState({serverError: err.message, isSaving: false});
-                }
-            });
+        const updateProgress = this.props.actions.updateMe({notify_props: data} as any);
+        Promise.resolve(updateProgress).then(({data: result, error: err}: any) => {
```

Comment:
I don't think this will work like you expect. `Promise.resolve` just returns a new promise that resolves to `updateProgress` which is a promise itself. That might get fixed if you change the type of `updateMe` like I mentioned above though